lint failure should not cause build to fail

**Is this a BUG REPORT or FEATURE REQUEST?**:
Build Issue:

If the `make lint` target fails the whole build is tagged as Failed. 
Currently make lint is only reporting warnings and as such is not really useful in the build process.
Ideally all the lint warnings should be fixed and then the make lint target should fail if we get any warnings from golint.

**What happened**:

`make lint` target failed as travis failed to pull github repository for golint. 

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

make lint target fails when travis fails to reach github
```
$ make lint
go get -v github.com/golang/lint/golint
github.com/golang/lint (download)
Fetching https://golang.org/x/lint?go-get=1
Parsing meta tags from https://golang.org/x/lint?go-get=1 (status code 200)
get "golang.org/x/lint": found meta tag main.metaImport{Prefix:"golang.org/x/lint", VCS:"git", RepoRoot:"https://go.googlesource.com/lint"} at https://golang.org/x/lint?go-get=1
golang.org/x/lint (download)
# cd .; git clone https://go.googlesource.com/lint /home/travis/gopath/src/golang.org/x/lint
Cloning into '/home/travis/gopath/src/golang.org/x/lint'...
fatal: remote error: Git repository not found
package golang.org/x/lint: exit status 128
make: *** [lint] Error 1
The command "make lint" exited with 2.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,8 +4,8 @@ services:
   - docker
 language: go
 go:
-  - 1.8.5
-  - 1.9.2
+  - 1.9.5
+  - 1.x
 before_install:
   - sudo apt-get update -yq
   - sudo apt-get -o Dpkg::Options::="--force-confnew" install -yq docker-ce
@@ -18,11 +18,8 @@ script:
   - go fmt $(go list ./... | grep -v vendor) | wc -l | grep 0
   - make install
   - make vet
-  - make lint
   - make docker-test
   - make docs
-  - make docker-coverage
-  - goveralls -coverprofile=/mnt/coverage.out -service=travis-ci -repotoken "${COVERALLS_TOKEN}"
   - if [ "${TRAVIS_BRANCH}" == "master" ] && [ "${TRAVIS_PULL_REQUEST}" == "false" ]; then
       make docker-build-osd;
       docker login -u "${DOCKER_USER}" -p "${DOCKER_PASS}";

